app/modules/booking/service.server.ts

```diff
@@ -197,11 +197,6 @@ export async function upsertBooking(
         data.custodianUser = {
           connect: { id: custodianUser.user.id },
         };
-      } else {
```

Comment:
This shouldnt be removed, we have to update the check. Currently if you have a booking with a user assigned and then change it to a team member, the user doesnt get disconnected. So I think we have to make sure that the custodianUser gets disconnected if `custodianTeamMemberId` is being sent.